src/fhirtypes/ElementDefinition.ts

```diff
@@ -1364,18 +1372,32 @@ export class ElementDefinition {
       this.type.length === 1 &&
       (this.type[0].profile == null || this.type[0].profile.length <= 1)
     ) {
-      // If it has a profile, use that, otherwise use the code
-      const type = this.type[0].profile?.[0] ?? this.type[0].code;
-      const def = resolve(type);
-      if (def) {
-        const newElements = def.elements.slice(1).map(e => {
+      let newElements: ElementDefinition[] = [];
+      if (this.sliceName) {
+        // If the element is sliced, we first try to unfold from the SD itself
+        const slicedElement = this.slicedElement();
+        newElements = slicedElement.children().map(e => {
+          const eClone = e.clone();
+          eClone.id = eClone.id.replace(slicedElement.id, this.id);
+          eClone.structDef = this.structDef;
+          eClone.captureOriginal();
+          return eClone;
+        });
+      }
+      if (newElements.length === 0) {
+        // If it has a profile, use that, otherwise use the code
+        const type = this.type[0].profile?.[0] ?? this.type[0].code;
+        const def = resolve(type);
+        newElements = def?.elements.slice(1).map(e => {
           const eClone = e.clone();
           eClone.id = eClone.id.replace(def.type, `${this.id}`);
           eClone.structDef = this.structDef;
           // Capture the original so that diffs only show what changed *after* unfolding
           eClone.captureOriginal();
           return eClone;
         });
+      }
+      if (newElements.length > 0) {
```

Comment:
This changes the signature so now it might return `undefined`.  Was there any danger in the previous approach (which would return an empty list instead)?

Reply:
There is still a `return []` at the bottom of the function, it gets cut off by this view in GitHub but if you view the full file it is there.

Reply:
Oh!  OK.  Sorry I missed that!